src/EnergyPlus/SingleDuct.cc

```diff
@@ -6015,6 +6015,27 @@ namespace SingleDuct {
         }
     }
 
+    void setATMixerSizingProperties(int const &inletATMixerIndex, int const &controlledZoneNum, int const &curZoneEqNum)
+    {
+        int airLoopIndex = // find air loop associated with ATMixer
+            DataZoneEquipment::ZoneEquipConfig(controlledZoneNum).InletNodeAirLoopNum(SingleDuct::SysATMixer(inletATMixerIndex).CtrlZoneInNodeIndex);
+        bool SizingDesRunThisAirSys = false;                               // Sizing:System object found flag
+        CheckThisAirSystemForSizing(airLoopIndex, SizingDesRunThisAirSys); // check for Sizing:System object
+        if (SizingDesRunThisAirSys) {
+            int sysSizIndex =
+                UtilityRoutines::FindItemInList(FinalSysSizing(airLoopIndex).AirPriLoopName, SysSizInput, &SystemSizingInputData::AirPriLoopName);
+            if (sysSizIndex == 0) sysSizIndex = 1; // use first when none applicable
+            // set ATMixer outlet air conditions in ZoneEqSizing array
+            ZoneEqSizing(curZoneEqNum).ATMixerVolFlow = SingleDuct::SysATMixer(inletATMixerIndex).DesignPrimaryAirVolRate;
+            // This needs work.
```

Comment:
Does this still need work?

Reply:
Using the Cool/HeatSupTemp as the ATMixer supply temp assumes there is a coil in the main air loop, which is probably not the case for DOAS. So it could be the PreCool/HeatTemp that is used here. Or maybe the SAT is the correct number as long as the user knows that. So "This needs work" simply means this needs more thought to catch all corner cases.

Reply:
I guess that's still my question then.  Do you want try to catch more corner cases with this before you call it done?

Reply:
I'll check to see what variable is used to size reheat coils.

Reply:
The single duct TU's use TermUnitFinalZoneSizing(CurTermUnitSizingNum).DesHeatCoilInTempTU to size the coils re heating capacity. There is a corresponding DesCoolCoilInTempTU. I still don't think sizing accounts for when coils are not present in either the air loop or OA system. What I would feel very comfortable with is checking for cooling/heating air loop coils, then use SAT, and if not present cooling/heating OA coils, then use pre-cool/heat temps, and if not present to use weather for respective cooling or heating sizing. I know there is a function to test for OA coils, not sure about a function to test for coils in air loop. This can be a real mess when air loop coils are present since I would have to mine equipment models (e.g., Furnace) to see which coils are present. So my original thought was to drop this in but I could work on it more if needed.

Reply:
Interesting. On initial search, it looks like DesCoolCoilInTemp simply grabs the Sizing:System cooling supply temp (
But the for heating, DesHeatCoilInTemp calls a function [GetHeatingSATempForSizing](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/SimAirServingZones.cc#L7237) which does check for the presence of system coils.

Reply:
Also, I see that I made some stupid lines back in [this commit](https://github.com/NREL/EnergyPlus/commit/ad8586e37217a3448b3ea7d11ff7f2721b57d9e2#diff-f51329ac8013440dabda3acd6c2d2a9aR6143) for the air terminal sizing stuff. There are now several lines that simply assign a variable to itself, [here](https://github.com/NREL/EnergyPlus/blob/73281ffba303cbd55bfa8f4e7e5646843f39866d/src/EnergyPlus/SimAirServingZones.cc#L6372-L6373), [here](https://github.com/NREL/EnergyPlus/blob/73281ffba303cbd55bfa8f4e7e5646843f39866d/src/EnergyPlus/SimAirServingZones.cc#L6376-L6377), and [here](https://github.com/NREL/EnergyPlus/blob/73281ffba303cbd55bfa8f4e7e5646843f39866d/src/EnergyPlus/SimAirServingZones.cc#L6439-L6440). You may as well delete those lines while you're in here.

Reply:
Great, now the reheat coil inlet temp is probably wrong if there is only an OAHX. And in GetHeatingSATempForSizing why calculate w and H to get T instead of a simple mixing calc on T ? You get the same answer either way don't you? It's a straight line between the state points. Unless the psyc routines protect against exceeding the saturation curve? The RH function does, I don't think the others do. For my purposes I'd like to know if there is any coil in the air loop, not just water, elec and gas hanging on a branch.

    // Check whether there are Central Heating Coils in the Primary Air System
    for (AirLoopNum = 1; AirLoopNum <= NumPrimaryAirSys; ++AirLoopNum) {
        FoundCentralHeatCoil = false;
        for (BranchNum = 1; !FoundCentralHeatCoil && BranchNum <= PrimaryAirSystem(AirLoopNum).NumBranches; ++BranchNum) {
            for (CompNum = 1; !FoundCentralHeatCoil && CompNum <= PrimaryAirSystem(AirLoopNum).Branch(BranchNum).TotalComponents; ++CompNum) {
                CompTypeNum = PrimaryAirSystem(AirLoopNum).Branch(BranchNum).Comp(CompNum).CompType_Num;
                if (CompTypeNum == WaterCoil_SimpleHeat || CompTypeNum == Coil_ElectricHeat || CompTypeNum == Coil_GasHeat) {
                    FoundCentralHeatCoil = true;
                }
            } // end of component loop
        }     // end of Branch loop
        PrimaryAirSystem(AirLoopNum).CentralHeatCoilExists = FoundCentralHeatCoil;
    } // end of AirLoop loop

Reply:
Well, it's not always a straight line between state points, as you suggest, it's allowing for hitting the saturation line.
OAHX isn't accounted for in *any* of the sizing calcs yet, so I wouldn't inject that here. That should be a separate mod.